### DISCUSSION TITLE
How can I feed Nornir inventory from a Django3 model / database.

### DISCUSSION THREAD

**Discussion by schizo100** [nornir-automation/nornir#638] [Q&A] [answered]
Hello Legends, 
What's the easiest way to generate nornir inventory from Django model? 

Thank You.

**Comment by dbarrosop** [comment, marked as answer]
best way would be to create a custom inventory plugin. The documentation has some details and examples about ir and you can check any of the existing plugins too for reference

**Reply by schizo100** [reply]
Ok great found the documentation here. 
I was looking at version 3.0 and it does not have the link below. 

https://nornir.readthedocs.io/en/v2.4.0/howto/writing_a_custom_inventory_plugin.html

Thank you for your help.

**Reply by dbarrosop** [reply]
Note that documentation doesn't apply to nornir 3 and that's why it isn't present.

It is not a tutorial but the gist of what the plugin has to do is here:

https://nornir.readthedocs.io/en/latest/api/nornir/core/plugins/inventory.html

which links to the object that the plugin needs to return:  https://nornir.readthedocs.io/en/latest/api/nornir/core/inventory.html#nornir.core.inventory.Inventory

Here is the simple inventory for reference:

https://github.com/nornir-automation/nornir/blob/develop/nornir/plugins/inventory/simple.py#L67

And don't forget the plugin needs to be registered:

https://nornir.readthedocs.io/en/latest/plugins/index.html#registering-plugins

And yeah, a tutorial might be nice in case someone is inclined to contribute one :)
